### PR TITLE
Convert 'components/about_build_modal/about_build_modal' to typescript

### PR DESCRIPTION
**Pull request by vraravam**
#### Summary
Convert 'components/about_build_modal/about_build_modal' to typescript

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/12475

### CODE REVIEW

**Comment by hanzei** [comment]
@vraravam Heads up that there are two build issues:
```
components/about_build_modal/about_build_modal.test.tsx:150:45 - error TS2554: Expected 1 arguments, but got 0.

150         wrapper.find(Modal).first().props().onExited?.();
                                                ~~~~~~~~~~~~

  node_modules/@types/react-bootstrap/index.d.ts:42:13
    42   onExited?(node: HTMLElement): any;
                   ~~~~~~~~~~~~~~~~~
    An argument for 'node' was not provided.
```

**Comment by vraravam** [comment]
> @vraravam Heads up that there are two build issues:
> 
> ```
> components/about_build_modal/about_build_modal.test.tsx:150:45 - error TS2554: Expected 1 arguments, but got 0.
> 
> 150         wrapper.find(Modal).first().props().onExited?.();
>                                                 ~~~~~~~~~~~~
> 
>   node_modules/@types/react-bootstrap/index.d.ts:42:13
>     42   onExited?(node: HTMLElement): any;
>                    ~~~~~~~~~~~~~~~~~
>     An argument for 'node' was not provided.
> ```

@hanzei - tx, I do not know how to solve this issue. Any help is appreciated.

**Comment by devinbinnie** [comment]
> > @vraravam Heads up that there are two build issues:
> > ```
> > components/about_build_modal/about_build_modal.test.tsx:150:45 - error TS2554: Expected 1 arguments, but got 0.
> > 
> > 150         wrapper.find(Modal).first().props().onExited?.();
> >                                                 ~~~~~~~~~~~~
> > 
> >   node_modules/@types/react-bootstrap/index.d.ts:42:13
> >     42   onExited?(node: HTMLElement): any;
> >                    ~~~~~~~~~~~~~~~~~
> >     An argument for 'node' was not provided.
> > ```
> 
> @hanzei - tx, I do not know how to solve this issue. Any help is appreciated.

@vraravam According to the type for `Modal` in `react-bootstrap`, you need to provide an argument of type `node` to `onExited`.

`onExited?(node: HTMLElement): any;`

I think in this case you can just pass `null` or mock out an `HTMLElement` as we aren't using the parameter anyways.

**Comment by vraravam** [comment]
\update-branch

**Comment by vraravam** [comment]
Seems like the last review comments have been fixed, but there's no word yet on testing. should we abandon this PR?

**Comment by jasonblais** [comment]
Hi @vraravam, thanks for your contribution! Some of our team members have been away for the holidays, so reviews have been slower last two weeks. We'll pick this up soon :) 

cc @calebroseland @devinbinnie

**Review by devinbinnie — CHANGES_REQUESTED** [review]
Thanks for your contribution @vraravam!
I've added a few comments for you to review.

File: components/about_build_modal/about_build_modal.test.tsx [hunk 1]
```diff
@@ -5,27 +5,41 @@ import React from 'react';
...
+        Version: string;
+        BuildNumber: string;
+        SQLDriverName: string;
+        BuildHash: string;
+        BuildHashEnterprise: string;
+        BuildDate: string;
+        TermsOfServiceLink: string;
+        PrivacyPolicyLink: string;
+    } | null = null;
+    let license: {
```
[the remark this answers is not in the document]
Comment (vraravam) [reply components/about_build_modal/about_build_modal.test.tsx:38]: This doesn't seem to work - `License` has different fields.

File: components/about_build_modal/about_build_modal.test.tsx [hunk 2]
```diff
@@ -5,27 +5,41 @@ import React from 'react';
...
+        function mock() {
+            return new RealDate(date);
+        }
+        mock.now = () => date.getTime();
+        global.Date = mock as any;
     }
 
-    let config = null;
-    let license = null;
+    let config: {
```
[the remark this answers is not in the document]
Comment (vraravam) [reply components/about_build_modal/about_build_modal.test.tsx:27]: done in the next commit

File: components/about_build_modal/about_build_modal.tsx [hunk 3]
```diff
@@ -14,35 +13,41 @@ import {AboutLinks} from 'utils/constants';
...
+    onHide: () => void;
 
-        /**
-         * Function that is called when the modal is dismissed
-         */
-        onHide: PropTypes.func.isRequired,
+    /**
+     * Global config object
+     */
+    config: any;
```
[the remark this answers is not in the document]
Comment (vraravam) [reply components/about_build_modal/about_build_modal.tsx:26]: done in the next commit

File: components/about_build_modal/about_build_modal.tsx [hunk 4]
```diff
@@ -14,35 +13,41 @@ import {AboutLinks} from 'utils/constants';
...
 
-        /**
-         * Global config object
-         */
-        config: PropTypes.object.isRequired,
+    /**
+     * Global license object
+     */
+    license: any;
```
[the remark this answers is not in the document]
Comment (vraravam) [reply components/about_build_modal/about_build_modal.tsx:31]: This doesn't seem to work - License has different fields.

File: components/about_build_modal/about_build_modal.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (devinbinnie) [reply components/about_build_modal/about_build_modal.test.tsx:30]: You're right. Can we use `ClientLicense`? It's currently just an alias for `Record<string, string>`, but eventually we'll type it properly using that same name.

File: components/about_build_modal/about_build_modal.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (vraravam) [reply components/about_build_modal/about_build_modal.tsx]: done with your latest suggestions @devinbinnie

File: components/about_build_modal/about_build_modal.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (vraravam) [reply components/about_build_modal/about_build_modal.test.tsx]: done with your latest suggestions @devinbinnie

File: components/about_build_modal/about_build_modal.tsx [hunk 5]
```diff
@@ -14,35 +15,41 @@ import {AboutLinks} from 'utils/constants';
...
+
+    /**
+     * Function that is called when the modal is dismissed
+     */
+    onHide: () => void;
+
+    /**
+     * Global config object
+     */
+    config: Partial<ClientConfig>;
```
Comment (devinbinnie) [inline components/about_build_modal/about_build_modal.tsx:28]: Is the only reason we're using `Partial` here to facilitate the tests?
If so, I think we should mock out the other variables in the tests and use just the base `ClientConfig` and `ClientLicense` here.

Comment (vraravam) [reply components/about_build_modal/about_build_modal.tsx:28]: Not really - there's code in `mattermost-redux/selectors/entities/general..d.ts` which uses a `Partial<ClientConfig>` - and I didn't want to change it in this PR.

Comment (devinbinnie) [reply components/about_build_modal/about_build_modal.tsx:28]: Huh, odd. It looks like `license` isn't using partial in `general.d.ts` though so we can change that one.

Comment (vraravam) [reply components/about_build_modal/about_build_modal.tsx:28]: Can that be handled in a separate PR please? I dont want to open a can of worms with that single change.

**Review by calebroseland — APPROVED** [review]

File: components/about_build_modal/about_build_modal.tsx (hunk 5 above)
Comment (devinbinnie) [reply components/about_build_modal/about_build_modal.tsx:28]: I'm not suggesting making any changes in `mattermost-redux`, I was just saying we could change the field type from `Partial<ClientLicense>` to `ClientLicense`. Or is that not possible?

Comment (vraravam) [reply components/about_build_modal/about_build_modal.tsx:28]: done in the latest commit

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @vraravam!

**Review by furqanmlk — APPROVED** [review]
Working as expected
All related test cases passed ✅

**Comment by devinbinnie** [comment]
@vraravam Can you update the branch and resolve conflicts? I can merge after that.

**Comment by vraravam** [comment]
done @devinbinnie

**Pull request merged**